src/admin-page/__tests__/UsersTable.test.js

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { MockedProvider } from '@apollo/client/testing';
+import { fireEvent, screen } from '@testing-library/dom';
+import userEvent from '@testing-library/user-event';
+import UsersTable from '../UsersTable';
+import {
+  CREATE_USER, DELETE_USER, GET_USERS, UPDATE_USER,
+} from '../../sources';
+
+const mocks = [
+  {
+    request: {
+      query: GET_USERS,
+    },
+    result: {
+      data: {
+        users: [{
+          id: 1, username: 'auser', password: 'a', firstName: 'A', lastName: 'A', userRole: { name: 'user' },
+        },
+        {
+          id: 2, username: 'buser', password: 'b', firstName: 'B', lastName: 'B', userRole: { name: 'sys_admin' },
+        }],
+      },
+    },
+  },
+
+  {
+    request: {
+      query: CREATE_USER,
+      variables: {
+        username: 'cuser',
+        password: 'c',
+        firstName: 'C',
+        lastName: 'C',
+        userRole: 1,
+      },
+    },
+    result: {
+      data: {
+        createUser: {
+          __typename: 'User',
+          id: 3,
+          username: 'cuser',
+          password: 'c',
+          firstName: 'C',
+          lastName: 'C',
+          userRole: {
+            __typename: 'UserRole',
+            name: 'sys_admin',
+          },
+        },
+      },
+    },
+  },
+
+  {
+    request: {
+      query: UPDATE_USER,
+      variables: {
+        id: 2,
+        username: 'duser',
+        password: 'd',
+        firstName: 'D',
+        lastName: 'D',
+        userRole: {
+          id: 1,
+        },
+      },
+    },
+    result: {
```

Comment:
always add `__typedef` while mocking gql responses.

Reply:
added __typename to every mocked response, where necesarry for now